From 3ac6aa8008970b9cbafaa7a7134c9dc2f455d2f7 Mon Sep 17 00:00:00 2001
From: Giovanni Campagna <gcampagna@src.gnome.org>
Date: Tue, 15 Mar 2011 20:37:29 +0100
Subject: [PATCH] include: mark flags as such

If the /*< flags >*/ annotation is missing, enums are picked as
enumerations, so bindings won't allow passing combinations of them.

https://bugzilla.gnome.org/show_bug.cgi?id=643011
---
 include/NetworkManager.h |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/include/NetworkManager.h b/include/NetworkManager.h
index d8f29c5..d421aff 100644
--- a/include/NetworkManager.h
+++ b/include/NetworkManager.h
@@ -138,6 +138,7 @@ typedef enum {
 
 /* 802.11 Access Point flags */
 typedef enum {
+	/*< flags >*/
 	NM_802_11_AP_FLAGS_NONE    = 0x00000000,
 	NM_802_11_AP_FLAGS_PRIVACY = 0x00000001
 } NM80211ApFlags;
@@ -150,6 +151,7 @@ typedef enum {
  * information elements.
  */
 typedef enum {
+	/*< flags >*/
 	NM_802_11_AP_SEC_NONE            = 0x00000000,
 	NM_802_11_AP_SEC_PAIR_WEP40      = 0x00000001,
 	NM_802_11_AP_SEC_PAIR_WEP104     = 0x00000002,
@@ -183,6 +185,7 @@ typedef enum {
  * Bluetooth device.
  */
 typedef enum {
+	/*< flags >*/
 	NM_BT_CAPABILITY_NONE = 0x00000000,
 	NM_BT_CAPABILITY_DUN  = 0x00000001,
 	NM_BT_CAPABILITY_NAP  = 0x00000002,
@@ -205,6 +208,7 @@ typedef enum {
  * API.
  */
 typedef enum {
+	/*< flags >*/
 	NM_DEVICE_MODEM_CAPABILITY_NONE      = 0x00000000,
 	NM_DEVICE_MODEM_CAPABILITY_POTS      = 0x00000001,
 	NM_DEVICE_MODEM_CAPABILITY_CDMA_EVDO = 0x00000002,
-- 
1.7.4